types/mapbox-gl/index.d.ts

```diff
@@ -555,7 +555,7 @@ declare namespace mapboxgl {
 
         on<T extends keyof MapLayerEventType>(
             type: T,
-            layer: string,
+            layer: string | ReadonlyArray<string>,
```

Comment:
`ReadOnlyArray` is something that is not really relevant for the API consumer. Only for the actual implementation, which in this case it does not use Typescript at all.

But it does not harm I guess.